### DISCUSSION TITLE
Nimib Speaking Hours - February session - Sun 26th - 11-12.30 CET

### DISCUSSION THREAD

**Discussion by pietroppeter** [pietroppeter/nimib#178] [Nimib speaking hours]
**edit**: confirmed we meet on Sunday Feb 26th, here is a **calendar event link**: https://calndr.link/event/zoYH5Rjmpo
**edit 2**: the time in calendar event link was wrong (I had put in 10 CET instead of 11 CET), now I fixed it!

Hey @HugoGranstrom what about next weekend for our next nimib speaking hours?
I would say Sunday (Feb 26th) at 11 am CET?

Agenda:
- review recent changes
- next activities?
- move something in nimib-org?
- code of conduct for nimib-org
- ...

**Comment by HugoGranstrom** [comment]
Sunday should work for me 👌

I don't have much to add to the agenda, will see how far I come with the "easy" PRs I'm assigned to. And if we want to start migrating a repo to the nimib-org, we could start with nimiBoost as it's not as vital as the other ones of we screw up somehow.

**Reply by pietroppeter** [reply]
confirmed and added the calendar event link above

**Reply by pietroppeter** [reply]
btw @HugoGranstrom if you want to catch up on the circus monkey island reference here is a video: https://www.youtube.com/watch?v=N37E2Lrc9PU (original pixelated) https://youtu.be/whIe6wG2sjE?t=138 (subsequent more polished but less vintage version)

in that video me and you would be the Fettucini brothers trying to convince the casual user to be launched with a cannon... er to contribute to nimib ecosystem... :D

**Comment by HugoGranstrom** [comment]
I just noticed that the calender event says that we start 10:00 tomorrow, is that on purpose or some time zone thing when you created it?

**Reply by pietroppeter** [reply]
Argh! To be honest I do not remember if I meant 10 or 11 (CET). What would be your preference?

**Reply by HugoGranstrom** [reply]
You wrote 11am in your original post ;) I prefer 11 tbh, need some sleep 🤣

**Reply by pietroppeter** [reply]
fixed the calendar link

**Comment by pietroppeter** [comment]
we are in! come on and join us!

https://meet.jit.si/NimibSpeakingHours

**Comment by pietroppeter** [comment]
recap of what we (me and @HugoGranstrom) discussed in Sunday's speaking hours. We also had a special guest @FedericoCeratto 🥳 

here are the notes from the meeting, they are a bit raw if you have questions, feel free to ask:
- last meet was jan 29th, see https://github.com/pietroppeter/nimib/discussions/162 (actually during the meeting I thought last meeting was Jan 21...)
- review past month activities  
  - [3 nimib releases](https://github.com/pietroppeter/nimib/releases), [milestone progress](https://github.com/pietroppeter/nimib/milestones): 3/8
    - 0.3.6: nim v2 compatibility + improved CI
    - 0.3.7: "all code shall be highlighted" + nbShow
    - 0.3.8: fix escapeTags
  - [2 nimibook releases](https://github.com/pietroppeter/nimibook/releases) ([2/6 milestones](https://github.com/pietroppeter/nimibook/milestones)):
    - 0.3.0: finally completed doc reworking
    - 0.3.1: nimibook fixes motivated by phil's work on owlkettle's docs
  - important discussion on nimib container blocks: https://github.com/pietroppeter/nimib/pull/177
    - includes a OOP-inspired minib: https://gist.github.com/pietroppeter/493ff1b546f783d12c935e7fc1e25532
  - discussion on nbCodeSkip (created issue in nimib: https://github.com/pietroppeter/nimib/issues/183)
  - discussion of inline skipping inside of nbCode (from discord chat)
    - no macro based nbCode
    - possible solutions are:
      - custom nbCode adding not shown code manually
      - use nbCodeSkip and merge in some way
        - actually we could have a mergeCode container block
  - discuss mermaid in readme (we could have a markdown-powered idea)
  - nblog: basic theme activity https://github.com/pietroppeter/nblog/pull/16: 
    - first successful nimib -> json -> html (important milestone for SSG)
      - very tricky because we have an ugly current context object that we should remove
  - nimibex: a new chat idea https://github.com/nimib-org/nimibex/issues/11
  - minor doc update on p5nim
- move something to nimib-org?
  - we decided we will pick a moment where we have a bit of time (next weekend?) and move everything (nimib, nimibook, nblog, ...)
  - what do we need to change after moving (github redirect should take care of most of the stuff):
    - [ ] hard coded links in docs
    - [ ] packages.json
    - pietro: avoid working with a nimib fork or it will mess up github redirect!
- code of conduct
  - adopt contributor covenant https://www.contributor-covenant.org
  - add a personal note on why we think this is important
  - add contact points for who is in charge of enforcing (what? twitter? discord? fosstodon? probably better to avoid mail)
  - also interesting: https://github.com/nim-works/nimskull/blob/devel/CODE_OF_CONDUCT.md
  - also interesting: https://fosdem.org/2023/schedule/event/learned_leading_healthy_project/
- ideas
  - we could actually take more advantage of markdown processor (for highlighting, headings, and stuff like mermaid diagrams, references). we could open an issue (or a discussion) for this
  - we could have nimib speaking hours be a multi author blogpost on nblog
  - actually a better minib would be where all blocks are (data: JsonNode, blocks: seq[NbBlock])
    - this should be a content field in NbDoc that is the only thing we serialize to json
    - we should avoid duplication of data (e.g. blk.code and blk.context["code"] but also `source` is both in NbDoc than in its context)
    - take advantage of dot operators?
- next activities
  - review milestone issues
  - this month with nim: an update from nimib team? also mention nimib speaking hours? yes and yes
  - complete nblog basic theme activity
  - start with nimibex? yay! Hugo
  - make yamly public (not currently working on it)
  - work on SSG (notes from last meeting)
    - next step: port https://github.com/dmulholl/ivy to nim as learning experience
    - maybe try also porting: https://www.webdevdrops.com/en/build-static-site-generator-nodejs-8969ebe34b22/

**Reply by pietroppeter** [reply]
I plan to make the This month with Nim post by end of day today.

**Reply by HugoGranstrom** [reply]
If you want me to take a look at it, I will be able to look at it tonight. I'm a bit busy until then

**Reply by pietroppeter** [reply]
> If you want me to take a look at it, I will be able to look at it tonight. I'm a bit busy until then

Yeah I plan anyway to put a warning not to share until you have reviewed it. Don't worry if you are not able this evening (this month with Nim does not come out right away)
